### ISSUE TITLE
 Cannot set property 'patches' of undefined

### ISSUE THREAD

**Issue by SeyZ** [broccolijs/broccoli-funnel#83]
Hey, I'm using ember-cli which uses this library. During the build, I suddenly have this stacktrace (works before). Any idea?

```
The Broccoli Plugin: [Funnel: Funnel: index.html] failed with:
TypeError: Cannot set property 'patches' of undefined
    at Funnel.processFilters (/app/tmp/cache/node_modules/broccoli-funnel/index.js:308:33)
    at Funnel.build (/app/tmp/cache/node_modules/broccoli-funnel/index.js:233:10)
    at /app/tmp/cache/node_modules/broccoli-plugin/read_compat.js:61:34
    at tryCatch (/app/tmp/cache/node_modules/rsvp/dist/rsvp.js:538:12)
    at invokeCallback (/app/tmp/cache/node_modules/rsvp/dist/rsvp.js:553:13)
    at publish (/app/tmp/cache/node_modules/rsvp/dist/rsvp.js:521:7)
    at flush (/app/tmp/cache/node_modules/rsvp/dist/rsvp.js:2373:5)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
The broccoli plugin was instantiated at: 
    at Funnel.Plugin (/app/tmp/cache/node_modules/broccoli-plugin/index.js:7:31)
    at new Funnel (/app/tmp/cache/node_modules/broccoli-funnel/index.js:57:10)
    at EmberApp.index (/app/tmp/cache/node_modules/ember-cli/lib/broccoli/ember-app.js:641:15)
    at EmberApp.toArray (/app/tmp/cache/node_modules/ember-cli/lib/broccoli/ember-app.js:1603:10)
    at EmberApp.toTree (/app/tmp/cache/node_modules/ember-cli/lib/broccoli/ember-app.js:1626:30)
    at module.exports (/tmp/build_261851e7af050fd916277787d1c1c44c/ForestAdmin-forestadmin-ea054b32b7621ae713614798a568b506d70f08a4/ember-cli-build.js:83:14)
    at CoreObject.module.exports.Task.extend.setupBroccoliBuilder (/app/tmp/cache/node_modules/ember-cli/lib/models/builder.js:74:19)
    at CoreObject.module.exports.Task.extend.init (/app/tmp/cache/node_modules/ember-cli/lib/models/builder.js:54:10)
    at CoreObject.superWrapper [as init] (/app/tmp/cache/node_modules/core-object/lib/assign-properties.js:32:18)
    at CoreObject.Class (/app/tmp/cache/node_modules/core-object/core-object.js:32:33)
 !     Push rejected, failed to compile Ember CLI app.
 !     Push failed
```
Add Comment
```

**Comment by msvanvertloo** [comment]
It's due to a recent heimdalljs update (0.2.1 works fine, 0.2.2 doesn't).

**Comment by derekdowling** [comment]
Just started seeing this as well. @msvanvertloo any ideas on how to resolve this?

**Comment by mikebarr58** [comment]
I was able to get around this by just specifying "heimdalljs": "0.2.1" in the package.json, for whatever that's worth

**Comment by derekdowling** [comment]
Awesome @mikebarr58 thanks for confirming that this is indeed the problem, I have a PR ready, just need someone to approve or merge.

**Comment by hjdivad** [comment]
see https://github.com/ember-cli/ember-cli/issues/6435

tl;dr v0.2.2 is a bad release, fixed with v0.2.3 release
